relaton/tests/test_serializers.py

```diff
@@ -266,6 +266,34 @@ def test_build_refcontent_string_with_date_type_different_than_published(self):
             date.get(date.keys()[0]), data["date"][0]["value"].split("-")[0]
         )
 
+    def test_create_reference_for_internet_drafts(self):
+        """
+        InternetDrafts entries require the reference target
```

Comment:
It's not only for Internet-Drafts, but all IETF-originated bibliographic sources.

Reply:
Based on [this comment](https://github.com/ietf-tools/bibxml-service/issues/266#issuecomment-1228060552) I guess this issue is still on hold? @ronaldtse

Reply:
@stefanomunarini from the comment stated by @rjsparks
> To be clear - at some point in the future, we will probably also return to using the target attribute for references for I-Ds and RFCs[...]

I believe this means that the RFCs and RFC subseries will also be served with the `<format>` element instead of the `target=` attribute.

Let's go ahead with changing, for all bibliographic types, to use the `<format>` element if there are multiple formats. I believe they are either HTML or PDF.

Reply:
@ronaldtse what do you mean with `if there are multiple formats`? Current implementation simply replaces <target with <format if doctype == "internet-draft". 

From a little research, I noticed that RFC and RFC-subseries entries do not have a doctype attribute (see https://github.com/ietf-tools/relaton-data-rfcs/blob/e6b1d288d5416497fd268039e28b547dbca0d8f2/data/RFC4872.yaml and https://github.com/ietf-tools/relaton-data-rfcsubseries/blob/main/data/BCP0003.yaml as examples). 

What is the best way to determine if a document is RFC or RFC-subseries?

Reply:
What is the best way to to determine if a BibItem is RFC or RFC-subseries?

Reply:
@stefanomunarini I believe Ronald accepts using `<format>` for _all_ bibliographic items. No need to check if they are RFC or RFC subseries.

(FWIW, we use some ways to check for it where absolutely required, they are [brittle](https://github.com/ietf-tools/bibxml-service/blob/67a2d1e550ffb01ac86970e7bee7c0ad9a97981a/bibxml/xml2rfc_adapters.py#L536-L542) since standardizing identifiers is not really in scope of Relaton spec.

In any case we shouldn’t have to do it in relaton-py unless we have to. We’ll move `bibxml` serializer to BibXML service soon so this should be less of an issue afterwards.)

Reply:
> No need to check if they are RFC or RFC subseries.

Yes.

> In any case we shouldn’t have to do it in relaton-py unless we have to. We’ll move bibxml serializer to BibXML service soon so this should be less of an issue afterwards.

I'm not sure about this... we need to discuss?

Reply:
> I'm not sure about this... we need to discuss?

What we discussed with Stefano before:

I originally made `bibxml` serialized part of `relaton-py`, but it turns out to be pretty much useless if it does things by the spec. We end up adjusting it all the time to mimic old xml2rfc path response. It causes frequent relaton-py releases. For all intents and purposes, `bibxml` serializer is part of BibXML service, but located in a wrong package.

- It can just be made part of the service (or a standalone package, but I don’t see much point in that since it’s really tied to BibXML service).
- We can keep the existing `bibxml` serialized in relaton-py, and make it follow the spec, but we can’t use it in BibXML service anyway.
- Taking spec-conformant XML element tree from relaton-py’s `bibxml` and manipulating it to match xml2rfc paths (e.g., adding `<format>` element and so on), I suspect, will be inefficient (both in terms of development work and in terms of runtime performance) and error-prone. It would require iterating over XML elements, deleting and updating them, inserting new ones at correct locations, etc.

Reply:
> I believe this means that the RFCs and RFC subseries will also be served with the <format> element instead of the target= attribute.
and

> No need to check if they are RFC or RFC subseries.

Based on [this comment](https://github.com/ietf-tools/bibxml-service/issues/266#issuecomment-1230860069) this update should only be for IDs and not for *all* BibliographicItems. 

@ronaldtse I will rollback to only update InternetDrafts XML format.

@strogonoff can I proceed with updating the serializer, or are we going to move it to BibXML-service first, and then update its behaviour there?

Reply:
> this update should only be for IDs and not for all BibliographicItems.

Sorry for the confusion! @stefanomunarini no... we want to use `format` for ALL BibliographicItems. Please proceed to apply this FOR ALL.

> It causes frequent relaton-py releases and they tie up work on bibxml-service

Why is that a bad thing? On the other hand, you can imagine that relaton-py could be directly included in xml2rfc, which resolves this issue (I'm saying it is *possible* to do this).

Reply:
@ronaldtse does this mean that we replace `target` with `format` for all BibliographicItems? Am I understanding it correctly?